packages/openactive-integration-tests/test/features/leasing/anonymous-leasing/implemented/lease-opportunity-capacity-update-test.js

```diff
@@ -0,0 +1,207 @@
+const { expect } = require('chai');
+const chakram = require('chakram');
+const { FeatureHelper } = require('../../../../helpers/feature-helper');
+const { GetMatch, C1, Common } = require('../../../../shared-behaviours');
+const { FlowHelper } = require('../../../../helpers/flow-helper');
+const { RequestState } = require('../../../../helpers/request-state');
+
+/**
+ * @typedef {import('chakram').ChakramResponse} ChakramResponse
+ */
+
+FeatureHelper.describeFeature(module, {
+  testCategory: 'leasing',
+  testFeature: 'anonymous-leasing',
+  testFeatureImplemented: true,
+  testIdentifier: 'lease-opportunity-capacity-update',
+  testName: 'Leased spaces are unavailable for purchase by other users',
+  testDescription: 'When an opportunity is leased, the capacity is decremented',
+  testOpportunityCriteria: 'TestOpportunityBookableFiveSpaces',
+  controlOpportunityCriteria: 'TestOpportunityBookable',
+  multipleOpportunityCriteriaTemplate: opportunityType => [{
+    opportunityType,
+    opportunityCriteria: 'TestOpportunityBookableFiveSpaces',
+    primary: true,
+    control: false,
+  },
+  {
+    opportunityType,
+    opportunityCriteria: 'TestOpportunityBookable',
+    primary: false,
+    control: true,
+  }],
+},
+(configuration, orderItemCriteria, featureIsImplemented, logger, parentState, parentFlow) => {
+  /**
+   * @param {Number} expected
+   * @param {C1} stage
+   * @param {Function} responseAccessor
+   */
+  function itShouldHaveCapacity(expected, stage, responseAccessor) {
+    Common.itForOrderItemByControl(orderItemCriteria, parentState, stage, () => responseAccessor().body,
+      'should decrement remaining slots',
+      (feedOrderItem, responseOrderItem) => {
+        if (responseOrderItem && responseOrderItem.orderedItem['@type'] === 'Slot') {
+          expect(responseOrderItem).to.nested.include({
+            'orderedItem.remainingUses': expected,
+          });
+        } else {
+          expect(responseOrderItem).to.nested.include({
+            'orderedItem.remainingAttendeeCapacity': expected,
+          });
+        }
+      },
+      null, null); // no control
+  }
+
+  /**
+   * @param {C1} stage
+   * @param {() => ChakramResponse} responseAccessor This is wrapped in a
+   *   function because the actual response won't be available until the
+   *   asynchronous before() block has completed.
+   */
+  function itShouldReturn409Conflict(stage, responseAccessor) {
+    it('should return 409', () => {
+      stage.expectResponseReceived();
+      chakram.expect(responseAccessor()).to.have.status(409);
+    });
+  }
+
+  /**
+   * @param {RequestState} state
+   * @param {Number} count
+   */
+  function setOrderItemsOnState(state, count) {
+    const orderItems = parentState.orderItems.filter(oi => !oi['test:control']);
```

Comment:
Alternatively, you could add a config var to FeatureHelper to turn off controls for a particular test 🤔

Reply:
I think if I use `TestOpportunityBookableFiveSpaces` for the control, it's no longer a control, so I'll go with your second suggestion.

Reply:
If we set `controlOpportunityCriteria` to `TestOpportunityBookableFiveSpaces`, I don't think it would really be a control, and would hence be redundant. As for extending `FeatureHelper`, to be honest, I think just passing `null` into `itForOrderItemByControl` is cleaner. It's best to avoid having too many configuration options - it can get out of hand.

Reply:
> If we set controlOpportunityCriteria to TestOpportunityBookableFiveSpaces, I don't think it would really be a control, and would hence be redundant

I do agree with that

As for extending FeatureHelper, the problem is that the description in FeatureHelper is used to derive stats about the tests, and therefore it's helpful if it's as accurate as possible. It also acts as a kind of documentation of the test. So, if the description says "I use a TestOpportunityBookable as a control item", the analysis and the code reader are going to assume that to be true.

If the test cannot use multiple items (although I'm curious to know why as it seems like it should be possible), then a control is not used anyway, in which case I'd suggest:

1. Removing `controlOpportunityCriteria` from the FeatureHelper description
2. Using `Common. itForOrderItem` instead of `Common.itForOrderItemByControl`
3. Removing this line that filters out control items

Reply:
I reverted my changes to `itForOrderItemByControl` to support `null` controls, as they're no longer required.